### ISSUE TITLE
Not returning primary generated column value when cascade inserting

### ISSUE THREAD

**Issue by MichalLytek** [typeorm/typeorm#4628]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[x] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I have nullable OneToOne relations in my entity with cascade insert:

```ts
@Entity()
export default class DistributorTags {
  @PrimaryGeneratedColumn()
  id!: number;

  @ManyToOne(type => Distributor, distributor => distributor.tags, {
    nullable: false,
  })
  distributor!: Distributor;
  @Column()
  distributorId!: number;

  @ManyToOne(type => Staff)
  staff!: Staff;
  @Column()
  staffId!: number;

  @Column()
  dateIssued!: Date;

  @Column("int", { width: 4 })
  issueYear!: number;

  @JoinColumn()
  @OneToOne(type => DistributorTagsIssue, {
    nullable: true,
    eager: true,
    cascade: ["insert"],
  })
  blueTags?: DistributorTagsIssue;

  @JoinColumn()
  @OneToOne(type => DistributorTagsIssue, {
    nullable: true,
    eager: true,
    cascade: ["insert"],
  })
  blackTags?: DistributorTagsIssue;
}

@Entity()
export default class DistributorTagsIssue {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column("int")
  noIssued!: number;

  @Column("int")
  startNo!: number;

  @Column("int")
  endNo!: number;
}
```
When I try to insert the new data to database:
```ts
    this.distributorTagsRepository.create({
      staffId: currentUser!.staffId!,
      distributorId,
      dateIssued: getCurrentDate(),
      issueYear,
      tagsRequest,
      ...(tagsColor === TagColor.BLACK
        ? { blackTags: tagsIssue }
        : tagsColor === TagColor.BLUE
        ? { blueTags: tagsIssue }
        : {}),
    });
    await this.distributorTagsRepository.save(distributorTags);
```
the `distributorTags.id` property is undefined.

However when I remove the cascade data:
```ts
   this.distributorTagsRepository.create({
      staffId: currentUser!.staffId!,
      distributorId,
      dateIssued: getCurrentDate(),
      issueYear,
      tagsRequest,
    });
    await this.distributorTagsRepository.save(distributorTags);
```
it properly returns `distributorTags.id` property.

**Comment by DimaIT** [comment]
Related to #4090

**Comment by AlexMesser** [comment]
fixed in #7336
